src/api/c/memory.cpp

```diff
@@ -313,3 +317,439 @@ af_err af_get_mem_step_size(size_t *step_bytes) {
+    AF_CHECK(getMemoryManager(handle_).shutdown_fn(handle_));
+}
+
+af_buffer_info MemoryManagerFunctionWrapper::alloc(
+    const size_t size, bool user_lock, const unsigned ndims,
+    const dim_t *const dims, const unsigned element_size) {
+    af_buffer_info bufferInfo;
+    AF_CHECK(getMemoryManager(handle_).alloc_fn(
+        handle_, &bufferInfo, size, (int)user_lock, ndims, dims, element_size));
+    return bufferInfo;
+}
+
+size_t MemoryManagerFunctionWrapper::allocated(void *ptr) {
+    size_t out;
+    AF_CHECK(getMemoryManager(handle_).allocated_fn(handle_, &out, ptr));
+    return out;
+}
+
+void MemoryManagerFunctionWrapper::unlock(void *ptr, af_event e,
+                                          bool user_unlock) {
+    AF_CHECK(
+        getMemoryManager(handle_).unlock_fn(handle_, ptr, e, (int)user_unlock));
+}
+
+void MemoryManagerFunctionWrapper::signalMemoryCleanup() {
+    AF_CHECK(getMemoryManager(handle_).signal_memory_cleanup_fn(handle_));
+}
+
+void MemoryManagerFunctionWrapper::printInfo(const char *msg,
+                                             const int device) {
+    AF_CHECK(getMemoryManager(handle_).print_info_fn(
+        handle_, const_cast<char *>(msg), device));
+}
+
+void MemoryManagerFunctionWrapper::userLock(const void *ptr) {
+    AF_CHECK(getMemoryManager(handle_).user_lock_fn(handle_,
+                                                    const_cast<void *>(ptr)));
+}
+
+void MemoryManagerFunctionWrapper::userUnlock(const void *ptr) {
+    AF_CHECK(getMemoryManager(handle_).user_unlock_fn(handle_,
+                                                      const_cast<void *>(ptr)));
+}
+
+bool MemoryManagerFunctionWrapper::isUserLocked(const void *ptr) {
+    int out;
+    AF_CHECK(getMemoryManager(handle_).is_user_locked_fn(
+        handle_, &out, const_cast<void *>(ptr)));
+    return !!out;
+}
+
+void MemoryManagerFunctionWrapper::usageInfo(size_t *alloc_bytes,
+                                             size_t *alloc_buffers,
+                                             size_t *lock_bytes,
+                                             size_t *lock_buffers) {
+    // Not implemented in the public memory manager API, but for backward
+    // compatibility reasons, needs to be in the common memory manager interface
+    // so that it can be used with the default memory manager. Called from
+    // deviceMemoryInfo from a backend - throws so as to properly propagate
+    AF_ERROR(
+        "Device memory info/usage info not supported "
+        "for custom memory manager",
+        AF_ERR_NOT_SUPPORTED);
+}
+
+float MemoryManagerFunctionWrapper::getMemoryPressure() {
+    float out;
+    AF_CHECK(getMemoryManager(handle_).get_memory_pressure_fn(handle_, &out));
+    return out;
+}
+
+bool MemoryManagerFunctionWrapper::jitTreeExceedsMemoryPressure(size_t bytes) {
+    int out;
+    AF_CHECK(getMemoryManager(handle_).jit_tree_exceeds_memory_pressure_fn(
+        handle_, &out, bytes));
+    return !!out;
```

Comment:
double not?

Reply:
`int` to `bool`, but I'll just cast it :)